src/plccng/Scanner/matcher/matcher_test.py

```diff
@@ -0,0 +1,33 @@
+from ...load_spec.parse_spec import parse_lexical_spec
```

Comment:
We probably don't want to depend on parse_lexical_spec. The input we'll be given is pre-parsed and pre-validated, and is JSON. When JSON is loaded in Python using `json.load` or `json.loads` it will become a nested structure of dict, list, str, int, bool, etc. So from the perspective of the Matcher, a lexical spec looks like...

```python
spec = [
    {
        "type": "Token",
        "name": "MINUS",
        "regex": "-"
    }
]
spec[0]['name'] == 'MINUS'
```

There may be other fields, like Line, but I don't think you'll need them. If you do, great.